samples/exactlyOnceListen.js

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * This application demonstrates how to perform basic operations on
+ * schemas with the Google Cloud Pub/Sub API.
+ *
+ * For more information, see the README.md under /pubsub and the documentation
+ * at https://cloud.google.com/pubsub/docs.
+ */
+
+// This is a generated sample. Please see typescript/README.md for more info.
+
+'use strict';
+
+// sample-metadata:
+//   title: Listen with exactly-once delivery
+//   description: Listen for messages on an exactly-once delivery subscription.
+//   usage: node exactlyOnceListen.js <subscription-name-or-id>
+
+// [START pubsub_subscriber_exactly_once]
+/**
+ * TODO(developer): Uncomment this variable before running the sample.
+ */
+// const subscriptionNameOrId = 'YOUR_SUBSCRIPTION_NAME_OR_ID';
+
+// Imports the Google Cloud client library
+const {PubSub} = require('@google-cloud/pubsub');
+
+// Creates a client; cache this for further use
+const pubSubClient = new PubSub();
+
+async function listenOnSubscriptionWithExactlyOnceDelivery(
+  subscriptionNameOrId,
+  timeout
+) {
+  // References an existing subscription
+  const subscription = pubSubClient.subscription(subscriptionNameOrId);
+
+  // Create an event handler to handle messages
+  let messageCount = 0;
+  const messageHandler = async message => {
+    console.log(`Received message ${message.id}:`);
+    console.log(`\tData: ${message.data}`);
+    console.log(`\tAttributes: ${message.attributes}`);
+    messageCount++;
+
+    // Use `ackWithResponse()` instead of `ack()` to get a Promise that tracks
+    // the result of the acknowledge call. When exactly-once delivery is enabled
+    // on the subscription, the message is guaranteed not to be delivered again
+    // if the ack Promise resolves.
+    try {
+      // When the Promise resolves, the value is always AckResponses.Success,
+      // signaling that the ack was accepted.
+      await message.ackWithResponse();
```

Comment:
Can we explain for the users what happens if `ackWithResponse` is called without exactly once delivery? I assume that it will say that the promise will always return success.

Reply:
Yeah, it will. I like all the doc comments, I'll go back and try to improve that stuff.